provider/internal/image.go

```diff
@@ -350,17 +349,14 @@ func (i *Image) client(ctx context.Context, state ImageState, args ImageArgs) (C
 // authenticated.
 func (i *Image) Check(
 	ctx context.Context,
-	_ string,
-	_ resource.PropertyMap,
-	news resource.PropertyMap,
-) (ImageArgs, []provider.CheckFailure, error) {
-	args, failures, err := infer.DefaultCheck[ImageArgs](ctx, news)
+	req infer.CheckRequest,
+) (infer.CheckResponse[ImageArgs], error) {
+	args, failures, err := infer.DefaultCheck[ImageArgs](ctx, req.NewInputs)
 	if err != nil || len(failures) != 0 {
-		return args, failures, err
+		return infer.CheckResponse[ImageArgs]{Failures: failures, Inputs: args}, err
 	}
 
-	// :(
-	preview := news.ContainsUnknowns()
```

Comment:
Easy to add CheckRequest.Preview 👍

Reply:
No; I do believe that `--continue-on-error` produces unknowns in a non dry-run mode.